scripts/validate_docstrings.py

```diff
@@ -183,20 +183,28 @@ def validate_pep8(self):
             )
         )
 
-        application = flake8.main.application.Application()
-        application.initialize(["--quiet"])
-
+        error_messages = []
         with tempfile.NamedTemporaryFile(mode="w", encoding="utf-8") as file:
             file.write(content)
             file.flush()
-            application.run_checks([file.name])
-
-        # We need this to avoid flake8 printing the names of the files to
-        # the standard output
-        application.formatter.write = lambda line, source: None
-        application.report()
-
-        yield from application.guide.stats.statistics_for("")
+            # GH40784
+            cmd = ["flake8", "--quiet", "--statistics", file.name]
+            response = subprocess.run(cmd, capture_output=True, text=True)
+            stdout = response.stdout
+            # Remove file name from the start
+            stdout = stdout.replace(file.name, "")
+            # Remove the first and last elements since they are always empty str
+            messages = stdout.split("\n")[1:-1]
+            error_messages.extend(messages)
+
+        # Parse error message
+        for error_message in error_messages:
+            # Preserve whitespaces with a group
+            error_items = re.split(r"(\s+)", error_message)
+            error_count = int(error_items[0])
+            error_code = error_items[2]
+            message = "".join(error_items[4:])
```

Comment:
would something like
```
error_count, error_code, message = error_message.split(maxsplit=2)
```
work?

Reply:
Yes, it will work. 
Thank for the suggestion :slightly_smiling_face: